### ISSUE TITLE
ChromeCast getting media customData from MediaManager

### ISSUE THREAD

**Issue by bojant987** [google/shaka-player#1975]
I built a shaka receiver app which works like a charm when targeted from web shaka player based sender. 
But i'm having difficulties making it work with android sender. I'm still fairly new to ChromeCast platform, so not sure if i'm doing something wrong or it's simply not possible.
Hoping someone with more domain knowledge can help me out.

So, in android app we have code that sets custom data with MediaInfo.Builder, pretty much by the numbers code, like in the examples: https://developers.google.com/android/reference/com/google/android/gms/cast/MediaLoadOptions.Builder.html#setCustomData(org.json.JSONObject)
I figured i can get this data in receiver generally by instantiating `MediaManager` and then ```mediaManager.onLoad = function(data) { this.customData = data.media.customData  }```
However, with shaka receiver, i can't get customData this way, since instantiating `MediaManager` in app is not possible after `CastReceiverManager` call it's `start` method. which is handled by shaka before my receiver loads.
And i can't find that shaka exposes this in any way, in fact i can't find a trace of `MediaManager` in shaka code at all.
For shaka sender we use `appDataCallback`, but with android i have no idea how to do this properly.
So, i guess the question is: how do i get custom media data from android sender to shaka receiver?

**Comment by theodab** [comment]
We use the Google Cast Chrome Sender API library, which is something we load into our demo page. See [this page](https://developers.google.com/cast/docs/chrome_sender/integrate) of documentation; specifically, the "Load the library" section. Looking around online, it looks like the `MediaManager` is part of a deprecated SDK? I don't know very much about it, honestly.

If you can send messages to the receiver over specific namespaces, you could send your own `appData` messages. You'd want to send them over the namespace `urn:x-cast:com.google.shaka.v2`. To be recognized, the message should be formatted like...
```
"{"type":"appData","appData":"YOUR DATA HERE"}"
```
And of course, you will want to have the receiver app receive and process this data using the `appDataCallback`, as you have already discovered.

I'm fairly sure that will work, anyway. But people making their own custom cast senders to interface with our cast receiver wasn't really an expected use-case, so we didn't put any effort into testing it or making it easy or anything.

**Comment by bojant987** [comment]
Yeah, i kinda know about sending messages, that's what i ended up implementing in android sender.
I actually got it working both ways now, it was a silent error in my android app and it wouldn't send "LOAD" media message.

Btw, `MediaManager ` is part of v2 Cast API, so it is a bit outdated, but not deprecated. Shaka uses v2 of Cast API as far as i can see.

Thanks for taking the time to respond! Closing.
